### PR TITLE
improve test covr

### PR DESCRIPTION
**Pull request by misarb**
The pull request includes changes that augment the test coverage to `78.38%`, with a focus on the `graph_test` file.
The changes in the pull request encompass the following:

-    Addition of new test cases to cover  the graph module.

- Improvement of existing test cases to include edge cases and handle exceptions appropriately.

I have ensured that all tests pass successfully,

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 build/
+log/
 
 # IDE
 .vscode/
-.cache/
\ No newline at end of file
+.cache/
```
Comment (bobluppes) [inline .gitignore]: Was there a reason to change this file? I would propose to leave it out of the diff

File: test/graaflib/graph_test.cpp [hunk 2]
```diff
@@ -85,5 +116,91 @@ namespace graaf {
...
+        const auto vertex_id_1{graph.add_vertex(10)};
+        const auto vertex_id_2 = graph.vertex_count() + 1;
+
+        ASSERT_THROW({
+            try
+            {
+                // Call the add_edge function with a non-existing vertex ID
+                graph.add_edge(vertex_id_1, vertex_id_2, 100);
+                // If the above line doesn't throw an exception, fail the test
+                FAIL() << "Expected std::out_of_range exception, but no exception was thrown.";
```
Comment (bobluppes) [inline test/graaflib/graph_test.cpp:133]: [minor]:
Properly checking this in gtest is quite verbose 😬 

However, if you want you could save a line by removing the `FAIL` call, as the outer most `ASSERT_THROW` would fail the test in case the `add_edge` call does not throw. Same for the other two tests below.

```suggestion
                // Call the add_edge function with a non-existing vertex ID
                graph.add_edge(vertex_id_1, vertex_id_2, 100);
```

**Review by bobluppes — COMMENTED** [review]
Thanks a lot for working on this, looks great 😃 

I would not change the `.gitignore` in this PR, but once that it adressed I would be happy to merge. Also added a small comment on the exception tests, you can choose whether you want to address it or leave as-is.

Tagging #8 as this addresses part of the issue.

File: .gitignore (hunk 1 above)
Comment (bobluppes) [reply .gitignore]: Ah, I see, in that case could you please remove the changes to the `.gitignore` file? Then I can merge the PR.

Comment (misarb) [reply .gitignore]: yes of course

**Review by bobluppes — APPROVED** [review]
Thanks for addressing this, LGTM 👍🏻
